sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalog.scala

```diff
@@ -184,4 +184,9 @@ abstract class ExternalCatalog {
 
   def listFunctions(db: String, pattern: String): Seq[String]
 
+  // --------------------------------------------------------------------------
+  // Resources
+  // --------------------------------------------------------------------------
+
+  def addJar(path: String): Unit
```

Comment:
this also implies that `InMemoryCatalog` can't work if users specify a custom SerDe class in CREATE TABLE. Considering this, should we throw exception in `InMemoryCatalog.addJar`?